### PR TITLE
Shallow water compass1.0

### PR DESCRIPTION
**Pull request by mark-petersen**
Adding shallow water core and single test case, Williamson case 1 cosine bell advection. This will be useful for projects using the shallow water core, including local time stepping and experimental mesh configurations.

### CODE REVIEW

**Comment by mark-petersen** [comment]
@xylar, I was working on partition tests with @gcapodag, and I thought it was worth the time to make a reproducible test in COMPASS, and learn it at the same time.

I copied the files from `compass/landice/dome` and made the obvious changes. Could you help me with this? When I try 
```
cd compass/
conda activate test_compass_1.0
python -m compass list
```
on the head of this repo, I see ocean cases, but sw and landice cases do not appear.

**Comment by xylar** [comment]
@mark-petersen, this is great!  I couldn't reproduce the behavior you describe on my laptop.  I see:

```
$ python -m compass list
Traceback (most recent call last):
  File "/home/xylar/miniconda3/envs/test_mpas/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/xylar/miniconda3/envs/test_mpas/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/xylar/code/mpas-work/compass/shallow_water_compass1.0/compass/__main__.py", line 7, in <module>
    from compass.list import main as main_list
  File "/home/xylar/code/mpas-work/compass/shallow_water_compass1.0/compass/list.py", line 7, in <module>
    from compass.testcases import collect
  File "/home/xylar/code/mpas-work/compass/shallow_water_compass1.0/compass/testcases.py", line 7, in <module>
    from compass.sw import tests as sw_tests
  File "/home/xylar/code/mpas-work/compass/shallow_water_compass1.0/compass/sw/tests/__init__.py", line 1, in <module>
    from compass.sw.tests import case_1_cosine_bell_advection
  File "/home/xylar/code/mpas-work/compass/shallow_water_compass1.0/compass/sw/tests/case_1_cosine_bell_advection/__init__.py", line 2, in <module>
    from compass.landice.tests.case_1_cosine_bell_advection import smoke_test, decomposition_test, \
ModuleNotFoundError: No module named 'compass.landice.tests.case_1_cosine_bell_advection'
```
I will see if it is somehow an issue with `test_compass_1.0` being out of date.  Was this on Grizzly or Anvil?  I'm going to try both, just in case.

**Comment by xylar** [comment]
On Anvil, I'm seeing:
```
$ source /lcrc/soft/climate/e3sm-unified/base/etc/profile.d/conda.sh
$ conda activate test_compass_1.0
$ python -m compass list
...
ImportError: cannot import name 'get_aggregator_by_name' from 'geometric_features.aggregation' (/lcrc/soft/climate/e3sm-unified/base/envs/test_compass_1.0/lib/python3.8/site-packages/geometric_features/aggregation/__init__.py)
```
So that means the conda environment is too far out of date.  The same is almost certainly true on Grizzly.  So I'm guessing you made your own environment.  Could you send me the output of the following?
```
conda list
conda info
```

**Comment by xylar** [comment]
I see now.  I didn't read your instructions carefully.  You're running:
```
cd compass/
conda activate test_compass_1.0
python -m compass list
```
You definitely don't want to do that first step.  When you do that, you're inside the `compass` package and there is no longer a local `compass/` directory for `python -m compass` to point to.  Since the way that the `test_compass_1.0` environment was created was to install `compass` as a package into the environment (but the version at that time), `python -m compass` will pick up the version from the envrionment if there isn't a local `compass` directory.

So leave out the `cd compass` and just do:
```
conda activate test_compass_1.0
python -m compass list
```
That wouldn't have worked yesterday anyway because the environment is out of date.  But I'm updating it now.

File: compass/sw/tests/case_1_cosine_bell_advection/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from compass.testcase import add_testcase
+from compass.landice.tests.case_1_cosine_bell_advection import smoke_test, decomposition_test, \
```
Comment (xylar) [inline compass/sw/tests/case_1_cosine_bell_advection/__init__.py]:
```suggestion
from compass.sw.tests.case_1_cosine_bell_advection import smoke_test, decomposition_test, \
```

**Comment by xylar** [comment]
Once I fix the issue above, I see:
```
$ python -m compass list
...
  86: sw/case_1_cosine_bell_advection/2000m/decomposition_test
  87: sw/case_1_cosine_bell_advection/2000m/restart_test
  88: sw/case_1_cosine_bell_advection/variable_resolution/smoke_test
  89: sw/case_1_cosine_bell_advection/variable_resolution/decomposition_test
  90: sw/case_1_cosine_bell_advection/variable_resolution/restart_test
```

**Comment by xylar** [comment]
The `test_compass_1.0` environments on Grizzly and Anvil are now up to date.

**Pull request closed without merging**